### PR TITLE
 nanonote: init at 1.4.1

### PR DESCRIPTION
**Pull request by Morgenkaff**
## Description of changes

Added nanonote to nixpkgs.
Git repo: https://github.com/agateau/nanonote

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/office/nanonote/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  ];
+
+  buildPhase = ''
+    cmake ..
+  '';
+
+  installPhase = ''
+    make install PREFIX=$out
+  '';
```
Comment (Janik-Haag) [inline pkgs/applications/office/nanonote/default.nix:41]: Please add preBuild/postBuild and preInstall/postInstall hooks using runhook here. take `pkgs/applications/misc/tandoor-recipes/frontend.nix` as a example

File: pkgs/applications/office/nanonote/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  meta = {
+    description = "A minimalist note taking application";
+    downloadPage = "https://github.com/agateau/nanonote";
```
Comment (Janik-Haag) [inline pkgs/applications/office/nanonote/default.nix:45]:
```suggestion
    downloadPage = "https://github.com/agateau/nanonote/releases/tag/${finalAttrs.version}";
```

File: pkgs/applications/office/nanonote/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    homepage = "https://github.com/agateau/nanonote";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ morgenkaff ];
+    platforms = lib.platforms.all;
```
Comment (Janik-Haag) [inline pkgs/applications/office/nanonote/default.nix:49]:
```suggestion
```
that's implicit so we don't need to specify it

**Review by Janik-Haag — COMMENTED** [review]
Can you squash all the update commits created by github into [nanonote: init at 1.4.1](https://github.com/NixOS/nixpkgs/pull/282307/commits/10b64531b1bf6787755476ea9aa0ae32edda1d57)? Other then that the preBuild/postBuild hook thing is still open, feel free to ask if you need some help/guidance

**Comment by Morgenkaff** [comment]
> Can you squash all the update commits created by github into [nanonote: init at 1.4.1](https://github.com/NixOS/nixpkgs/pull/282307/commits/10b64531b1bf6787755476ea9aa0ae32edda1d57)? Other then that the preBuild/postBuild hook thing is still open, feel free to ask if you need some help/guidance

Sorry for the late reply..
I thought I could manage to squash the commits (still new to the more advanced use of git).
But every time I try to squash the update commits, I have to merge. Is that the way it works? If not, can you help me?
Either just point me to the correct phrase to search for, or command to use/read up on.

Thank you!

**Comment by Morgenkaff** [comment]
Closing this.
I have no intention to become part of or contribute to the nix community anymore.

**Pull request closed without merging**